src/Microsoft.Diagnostics.Tools.RuntimeClient/Extensions.cs

```diff
@@ -21,6 +21,16 @@ public static void WriteString(this BinaryWriter @this, string value)
                 @this.Write(Encoding.Unicode.GetBytes(value + '\0'));
         }
 
+        public static void WriteAsciiString(this BinaryWriter @this, string value)
```

Comment:
Does the string need to be ascii? It would be nice to keep the protocol consistent on the string format we are already using (I assume Encoding.Unicode above means UTF16 but the API name is a bit ambiguous)

Reply:
It makes the runtime side so much easier because I would have to convert the utf string to ascii on native code because the PAL createdump launch code takes the path as ascii.

Reply:
Doing the conversion should be [easy](https://github.com/dotnet/coreclr/blob/master/src/inc/utilcode.h#L300)

    MAKE_UTF8PTR_FROMWIDE(pUtf8String, pUTF16String);

(I'm pretty sure execve takes UTF8 rather than truly only accepting ascii)

The protocol is like a public API and I'd like to keep it consistent as long as it isn't too onerous. I don't think this one will be.

Reply:
Never mind the comment about execve accepting UTF8 or not.  The macro uses WideCharToMultiByte which is used all over the PAL to convert to char* (UTF8).